tests/helpers/stat_tests.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Helper functions for testing stochastic processes.
+"""
+import numpy as np
+from scipy.stats import fisher_exact
+
+
+def fisher_exact_test(actual, expected, outcomes=(0, 1), threshold=0.1):
```

Comment:
Nice choice of stat test! I reckon this might not be ideal for dealing with all integration & system stochastic tests as it strictly checks the validity of `p_value` but it should satisfy the needs for one-shot measurement tests when returning `qml.sample`. We can always get back to this and add/revise different stats in `stat_tests.py` in the future.